include/RAJA/util/View.hpp

```diff
@@ -111,6 +111,158 @@ struct View {
+RAJA::Index_type selecttuple( Lay lyout, Tup&& tup, camp::idx_seq<Idxs...> )
+{ 
+  return lyout(
+                camp::get<Idxs>(std::forward<Tup>(tup))...
+              );
+}
+
+// sequence combiner
+template <typename Seq1, typename Seq2>
+struct cat_seq;
+
+template <camp::idx_t... Idxs1, camp::idx_t... Idxs2>
+struct cat_seq < camp::idx_seq<Idxs1...>,
+                      camp::idx_seq<Idxs2...>
+                    >
+{
+  using type = camp::idx_seq<Idxs1..., Idxs2...>;
+};
+
+template <typename Seq1, typename Seq2>
+using cat_seq_t = typename cat_seq<Seq1, Seq2>::type;
+
+// sequence offsetter
+template <camp::idx_t Offset, typename Seq>
+struct offset_seq;
+
+template <camp::idx_t Offset, camp::idx_t... Idxs>
+struct offset_seq<Offset, camp::idx_seq<Idxs...>>
+{
+  using type = camp::idx_seq<(Idxs+Offset)...>;
+};
+
+template <camp::idx_t Offset, typename Seq>
+using offset_seq_t = typename offset_seq<Offset, Seq>::type;
+
+// remove the Nth index in a parameter pack
+// returns linear index of layout(ar...)
+template <typename Lay, RAJA::Index_type Nth = 0, typename Tup>
+RAJA_HOST_DEVICE RAJA_INLINE auto removenth( Lay lyout, Tup&& tup ) ->
+  decltype( selecttuple<Lay>(
+              lyout,
+              std::forward<Tup>(tup),
+              cat_seq_t<  camp::make_idx_seq_t<Nth>,  // sequence up to Nth
+                          offset_seq_t<
+                            Nth+1,
+                            camp::make_idx_seq_t<camp::tuple_size<Tup>::value - Nth-1>
+                          > // sequence after Nth
+                       >{}
+            )
+          )
+{
+  return selecttuple<Lay>(
+              lyout,
+              std::forward<Tup>(tup),
+              cat_seq_t<  camp::make_idx_seq_t<Nth>,  // sequence up to Nth
+                          offset_seq_t<
+                            Nth+1,
+                            camp::make_idx_seq_t<camp::tuple_size<Tup>::value - Nth-1>
+                          > // sequence after Nth
+                       >{}
+          );
+}
+
+
+
+
+// P2Pidx represents the pointer-to-pointer index for swizzling of indices.
+// Default of 0 means that the p2p index is in the 0th position.
+template <typename ValueType,
+          typename LayoutType,
+          RAJA::Index_type P2Pidx = 0,
+          typename PointerType = ValueType **>
+struct MultiView {
+  using value_type = ValueType;
+  using pointer_type = PointerType;
+  using layout_type = LayoutType;
+  using nc_value_type = typename std::remove_const<value_type>::type;
+  using nc_pointer_type = 
+    typename std::add_pointer<
+      typename std::add_pointer<
+        typename std::remove_const<
+          typename std::remove_pointer<
+            typename std::remove_pointer<pointer_type>::type
+          >::type
+        >::type
+      >::type
+    >::type;
+  using NonConstView = MultiView<nc_value_type, layout_type, P2Pidx, nc_pointer_type>;
+
+  layout_type const layout;
+  nc_pointer_type data;
```

Comment:
@artv3 @rhornung67 or anyone else who uses views/layouts: I had to make the pointer of pointers data non-const in order to be able to construct a "const MultiView" from a non-const MultiView.  This differs from the original View implementation which preserves the const-ness of the data. Do you think this might be a problem?

Reply:
@rchen20 I don't think so if users are doing the right thing. @ajkunen thoughts on this?